ddht/v5_1/alexandria/broadcast_log.py

```diff
@@ -0,0 +1,37 @@
+import hashlib
+import time
+from typing import Dict
+
+from eth_typing import Hash32, NodeID
+from lru import LRU
+
+from ddht.v5_1.alexandria.advertisements import Advertisement
+from ddht.v5_1.alexandria.constants import ONE_HOUR
+
+
+class BroadcastLog:
+    """
+    Tracks the last time each advertisement was broadcast to each peer.
+    """
+
+    def __init__(self, max_records: int = 8192):
+        self._log: Dict[Hash32, int] = LRU(max_records)
+
+    def log(self, node_id: NodeID, advertisement: Advertisement) -> None:
```

Comment:
curious if recent thoughts on `ACK` affect how you're thinking about these logs, e.g., if no `ACK` was received, do we still log the broadcast/not ping them again?

Reply:
If we didn't receive an `Ack` then the call to `network.advertise(...)` would raise a `trio.TooSlow` and so we wouldn't end up making the call to `BroadcastLog.log(...)` so, no, it would not get broadcast and on some subsequent *loop* over the advertisements we'd re-broadcast, yes.  (this is just how the code is written today)

https://github.com/ethereum/ddht/blob/6bb7cb76afb8f9902f929f309d3feece6e1e1d90/ddht/v5_1/alexandria/advertisement_manager.py#L124-L136